### PR TITLE
Create Test Driven Development Project

### PR DESCRIPTION
**Pull request by karelvanoordt**
- Create test for the following methods:
-- `factorial`
-- `reverse`
-- `fizzbuzz`

- Add error test cases
- Build `Solver` class and its methods

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    buzz = @solver.fizzbuzz(10)
+    expect(buzz).to eq('buzz')
+  end
+
+  ## fizzbuzz method validation 3
+  it 'for a number divisible by 3 and 5, should return fizzbuzz' do
+    fizzbuzz = @solver.fizzbuzz(15)
+    expect(fizzbuzz).to eq('fizzbuzz')
+  end
+end
```
Comment (julie-ify) [inline spec/solver_spec.rb]: - [ ] Let's add a test that checks for when a number is not divisible by 3 or 5. An example of such a number is `7`.

File: solver.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def fizzbuzz(num)
+    return 'fizzbuzz' if (num % 3).zero? && (num % 5).zero?
+    return 'fizz' if (num % 3).zero?
+    return 'buzz' if (num % 5).zero?
+
+    num.to_s
+  end
+end
```
Comment (julie-ify) [inline solver.rb:20]: - [ ] Clean code 💯

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @karelvanoordt and @ShahierNashaat,

Good job so far!

## Highlight!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- TDD was followed ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: solver.rb (hunk 2 above)
Comment (ShahierNashaat) [reply solver.rb:20]: Hey @julie-ify what do you mean by clean code?

Comment (julie-ify) [reply solver.rb:20]: Your code is clean and easy to understand. Keep it up 💪

Comment (ShahierNashaat) [reply solver.rb:20]: Thank you so much @julie-ify for your feedback.

**Review by dasileker — APPROVED** [review]
## **Hi @karelvanoordt,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**